API/Services/StreamService.cs

```diff
@@ -344,4 +349,57 @@ public async Task DeleteExternalSource(int userId, int externalSourceId)
 
         await _unitOfWork.CommitAsync();
     }
+
+    public async Task DeleteSideNavSmartFilterStream(int userId, int sideNavStreamId)
+    {
+        var streams2 = await _unitOfWork.UserRepository.GetSideNavStreamWithUser(sideNavStreamId);
+        if (streams2 == null) throw new KavitaException("sidenav-stream-doesnt-exist");
+
+        if (streams2.AppUser.Id != userId) throw new KavitaException("sidenav-stream-doesnt-exist");
+
+
+        if (streams2.StreamType != SideNavStreamType.SmartFilter)
+        {
+            throw new KavitaException("sidenav-stream-only-delete-smart-filter");
+        }
+
+        _unitOfWork.UserRepository.Delete([streams2]);
+
+        await _unitOfWork.CommitAsync();
+    }
+
+    public async Task DeleteDashboardSmartFilterStream(int userId, int dashboardStreamId)
+    {
+        var streams2 = await _unitOfWork.UserRepository.GetDashboardStreamWithUser(dashboardStreamId);
+        if (streams2 == null) throw new KavitaException("dashboard-stream-doesnt-exist");
+
+        if (streams2.AppUser.Id != userId) throw new KavitaException("sidenav-stream-doesnt-exist");
+
+        if (streams2.StreamType != DashboardStreamType.SmartFilter)
```

Comment:
I'm wondering why we need to have 2 different methods and we can't just handle in one DeleteSmartFilter method?